In [1]:
from notebooks._utils import calculate_series_ensemble_accuracy

ds_name = "myriadlama"
dataset_root = "/home/xzhao/workspace/GYB_self-ensemble/datasets"

In [2]:
from notebooks._utils import calculate_parallel_ensemble_accuracy


for model_name in ["llama3.2_3b", "llama3.2_3b_it", "qwen2.5_3b", "qwen2.5_3b_it", "qwen2.5_7b", "qwen2.5_7b_it"]:
# for model_name in ["llama3.2_1b"]:
    print(f"\n=================== Model: {model_name} ===================")
    dump_file_prefix = f"{dataset_root}/{ds_name}/{model_name}/myriadlama."
    
    print("---- Calculating baseline ----")
    calculate_series_ensemble_accuracy(
        dump_file_prefix=dump_file_prefix, 
        single_para_qapair=True, explicit_prompts=False, repeat_paras=False,
        modifyattn=False, modifyrope=False, scale_score=0, 
        num_paraphrases=1, num_fewshots=5, use_generation=True)

    print("\n---- ❌ attention mask / ❌ rope modifications / ❌ scale score ----")
    calculate_series_ensemble_accuracy(
        dump_file_prefix=dump_file_prefix, 
        single_para_qapair=True, explicit_prompts=False, repeat_paras=False,
        modifyattn=False, modifyrope=False, scale_score=0, 
        num_paraphrases=5, num_fewshots=5, use_generation=True)
    
    
    print("\n---- ✅ Attention mask / ✅ Rope modifications / ❌ scale score ----")
    calculate_series_ensemble_accuracy(
            dump_file_prefix=dump_file_prefix, 
            single_para_qapair=True, explicit_prompts=False, repeat_paras=False,
            modifyattn=True, modifyrope=True, scale_score=0, 
            num_paraphrases=5, num_fewshots=5, use_generation=True)
    
    print("\n---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score ----")
    calculate_series_ensemble_accuracy(
        dump_file_prefix=dump_file_prefix, 
        single_para_qapair=True, explicit_prompts=False, repeat_paras=False,
        modifyattn=True, modifyrope=True, scale_score=True,
        num_paraphrases=5, num_fewshots=5, use_generation=True)

    print("\n---- Parallel Ensemble (avg logits) ----")
    calculate_parallel_ensemble_accuracy(
            dump_file_prefix=dump_file_prefix, 
            repeat_paras=False,
            num_paraphrases=5, num_fewshots=5,
            logits_ensemble_method="avg",
            use_generation=True)
            


=================== Model: llama3.2_3b ===================
---- Calculating baseline ----
Acc: 0.4808 ==> 🏷️ 1paras 5shots 1QA      (Baseline)

---- ❌ attention mask / ❌ rope modifications / ❌ scale score ----
Acc: 0.5457 ==> 🏷️ 5paras 5shots 1QA     

---- ✅ Attention mask / ✅ Rope modifications / ❌ scale score ----
Acc: 0.5557 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope 

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score ----
File ./modifyattn.modifyrope.scalescore.singleparaqapair.5samples.5paras.feather does not exist!

---- Parallel Ensemble (avg logits) ----
Acc: 0.5769 ==> 🏷️ 5paras 5shots  None layerNone  alpha1.0 token-all

=================== Model: llama3.2_3b_it ===================
---- Calculating baseline ----
Acc: 0.4050 ==> 🏷️ 1paras 5shots 1QA      (Baseline)

---- ❌ attention mask / ❌ rope modifications / ❌ scale score ----
Acc: 0.4720 ==> 🏷️ 5paras 5shots 1QA     

---- ✅ Attention mask / ✅ Rope modifications / ❌ scale score ----
Acc: 0.4721 ==> 🏷️ 5paras 5sho


=================== Model: llama3.2_3b ===================
---- Calculating baseline ----
Acc: 0.4640 ==> 🏷️ 1paras 5shots 1QA      (Baseline)

---- ❌ attention mask / ❌ rope modifications / ❌ scale score ----
Acc: 0.5330 ==> 🏷️ 5paras 5shots 1QA     

---- ✅ Attention mask / ✅ Rope modifications / ❌ scale score ----
Acc: 0.5320 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope 

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.2 ----
Acc: 0.5340 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.4 ----
Acc: 0.5350 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.6 ----
Acc: 0.5350 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.8 ----
Acc: 0.5360 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- Parallel Ensemble (avg logits) ----
Acc: 0.5640 ==> 🏷️ 5paras 5shots  None layerNone  alpha1.0 token-all

=================== Model: qwen2.5_3b ===================
---- Calculating baseline ----
Acc: 0.4090 ==> 🏷️ 1paras 5shots 1QA      (Baseline)

---- ❌ attention mask / ❌ rope modifications / ❌ scale score ----
Acc: 0.3870 ==> 🏷️ 5paras 5shots 1QA     

---- ✅ Attention mask / ✅ Rope modifications / ❌ scale score ----
Acc: 0.4500 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope 

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.2 ----
Acc: 0.4510 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.4 ----
Acc: 0.4530 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.6 ----
Acc: 0.4520 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.8 ----
Acc: 0.4570 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- Parallel Ensemble (avg logits) ----
Acc: 0.4660 ==> 🏷️ 5paras 5shots  None layerNone  alpha1.0 token-all

=================== Model: llama3.2_3b_it ===================
---- Calculating baseline ----
Acc: 0.4030 ==> 🏷️ 1paras 5shots 1QA      (Baseline)

---- ❌ attention mask / ❌ rope modifications / ❌ scale score ----
KeyError: 'generation_lemmas' ==> 🏷️ 5paras 5shots 1QA     

---- ✅ Attention mask / ✅ Rope modifications / ❌ scale score ----
KeyError: 'generation_lemmas' ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope 

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.2 ----
Acc: 0.4700 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.4 ----
Acc: 0.4730 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.6 ----
Acc: 0.4720 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.8 ----
Acc: 0.4790 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- Parallel Ensemble (avg logits) ----
Acc: 0.5060 ==> 🏷️ 5paras 5shots  None layerNone  alpha1.0 token-all

=================== Model: qwen2.5_3b_it ===================
---- Calculating baseline ----
Acc: 0.4070 ==> 🏷️ 1paras 5shots 1QA      (Baseline)

---- ❌ attention mask / ❌ rope modifications / ❌ scale score ----
Acc: 0.4230 ==> 🏷️ 5paras 5shots 1QA     

---- ✅ Attention mask / ✅ Rope modifications / ❌ scale score ----
File /home/xzhao/workspace/GYB_self-ensemble/datasets/myriadlama-debug/qwen2.5_3b_it/myriadlama.modifyattn.modifyrope.singleparaqapair.5samples.5paras.feather does not exist!

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.2 ----
Acc: 0.4570 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.4 ----
Acc: 0.4580 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.6 ----
Acc: 0.4620 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- ✅ Attention mask / ✅ Rope modifications / ✅ scale score: 0.8 ----
Acc: 0.4640 ==> 🏷️ 5paras 5shots 1QA   +Attn +Rope +ScaleScore

---- Parallel Ensemble (avg logits) ----
Acc: 0.4700 ==> 🏷️ 5paras 5shots  None layerNone  alpha1.0 token-all
